sos/report/plugins/cgroups.py

```diff
@@ -30,6 +30,13 @@ def setup(self):
         ])
 
         self.add_cmd_output("systemd-cgls")
+        self.add_forbidden_path([
+            "/sys/fs/cgroup/memory/memory.kmem.slabinfo",
+            "/sys/fs/cgroup/memory/*/memory.kmem.slabinfo",
+            "/sys/fs/cgroup/memory/*/*/memory.kmem.slabinfo",
+            "/sys/fs/cgroup/memory/*/*/*/memory.kmem.slabinfo",
+            "/sys/fs/cgroup/memory/*/*/*/*/memory.kmem.slabinfo"
+        ])
```

Comment:
Couldn't this be simplified to:

```python
self.add_forbidden_path("/sys/fs/cgroup/memory/*memory.kmem.slabinfo", recursive=True)
```

?

Reply:
It cannot, details below

```
# grep memory.kmem.slabinfo /var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sos_logs/sos.log
2022-03-22 08:05:39,031 INFO: [plugin:cgroups] adding forbidden path '/sys/fs/cgroup/memory/*memory.kmem.slabinfo'
...
```

```
# find /var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/ -iname "memory.kmem.slabinfo"
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/sshd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-remount-fs.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/import-state.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-disk-by\x2did-dm\x2duuid\x2dLVM\x2dUwicfx4ITiaIH0a5OsgOx3GvAQbA3EewXGEgwB55xJgWj51qhpW3wqMuQySfv2QB.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/NetworkManager.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/crond.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/kpatch.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/system-getty.slice/getty@tty1.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/system-getty.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-tmpfiles-setup.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-udevd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/proc-sys-fs-binfmt_misc.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-journald.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/-.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/kmod-static-nodes.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-disk-by\x2duuid-ac5451ff\x2d5c24\x2d4768\x2d8fac\x2dc96986b9630f.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/sssd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/lvm2-monitor.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/rhsmcertd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/sssd-kcm.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/system-sshd\x2dkeygen.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-sysctl.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/kdump.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-update-utmp.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-journal-flush.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/puppet.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dbus.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-journald.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dm-event.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/chronyd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-udev-trigger.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-coredump.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/rsyslog.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-random-seed.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/system-user\x2druntime\x2ddir.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/plymouth-read-write.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/plymouth-start.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-logind.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/atd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-dm\x2d1.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/boot.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-mqueue.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/rngd-wake-threshold.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/plymouth-quit.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-user-sessions.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/system-lvm2\x2dpvscan.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-udevd-kernel.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/system-systemd\x2dhibernate\x2dresume.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/rngd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-journald-dev-log.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dbus.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/sys-kernel-config.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/qemu-guest-agent.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/plymouth-quit-wait.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-mapper-rhel_jjansky\x2d\x2drhel8\x2dswap.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-tmpfiles-setup-dev.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/lvm2-lvmpolld.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/tuned.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/polkit.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dracut-shutdown.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/NetworkManager-wait-online.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-initctl.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-disk-by\x2did-dm\x2dname\x2drhel_jjansky\x2d\x2drhel8\x2dswap.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-rhel_jjansky\x2drhel8-swap.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/auditd.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/firewalld.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/dev-hugepages.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/sys-kernel-debug.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/systemd-udevd-control.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/sys-kernel-tracing.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/system.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-disk-by\x2duuid-ac5451ff\x2d5c24\x2d4768\x2d8fac\x2dc96986b9630f.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-hugepages.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/-.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-mapper-rhel_jjansky\x2d\x2drhel8\x2dswap.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dbus.socket/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/init.scope/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/proc-sys-fs-binfmt_misc.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/sys-kernel-config.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-disk-by\x2did-dm\x2duuid\x2dLVM\x2dUwicfx4ITiaIH0a5OsgOx3GvAQbA3EewXGEgwB55xJgWj51qhpW3wqMuQySfv2QB.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-mqueue.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-dm\x2d1.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/sys-kernel-tracing.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-disk-by\x2did-dm\x2dname\x2drhel_jjansky\x2d\x2drhel8\x2dswap.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/run-user-0.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/sys-kernel-debug.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/boot.mount/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/dev-rhel_jjansky\x2drhel8-swap.swap/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/user@0.service/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/session-7.scope/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/user-0.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/user.slice/memory.kmem.slabinfo
/var/tmp/sosreport-jjansky-rhel8-2022-03-22-ztpddfx/sys/fs/cgroup/memory/init.scope/memory.kmem.slabinfo
```

Reply:
Ah, my fault. For recursive globs it'd be `**`, e.g.:

```python
self.add_forbidden_path("/syus/fs/cgroup/memory/**/memory.kmem.slabinfo", recursive=True)
```

```
>>> glob.glob("/sys/fs/cgroup/memory/**/memory.kmem.slabinfo", recursive=True)
['/sys/fs/cgroup/memory/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/user.slice/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/user.slice/user-1000.slice/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/user.slice/user-1000.slice/session-17.scope/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/user.slice/user-1000.slice/user@1000.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/init.scope/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/systemd-update-utmp.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/systemd-sysusers.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/system-systemd\\x2dfsck.slice/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/system-systemd\\x2dfsck.slice/systemd-fsck@dev-disk-by\\x2duuid-7B77\\x2d95E7.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/systemd-udevd-control.socket/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/lvm2-monitor.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/systemd-journal-flush.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/systemd-sysctl.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/systemd-udevd.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/systemd-udevd-kernel.socket/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/system-serial\\x2dgetty.slice/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/system-serial\\x2dgetty.slice/serial-getty@ttyS0.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/ldconfig.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/rpc-statd-notify.service/memory.kmem.slabinfo', '/sys/fs/cgroup/memory/system.slice/import-state.service/memory.kmem.slabinfo',
[....]
'/sys/fs/cgroup/memory/system.slice/var-lib-nfs-rpc_pipefs.mount/memory.kmem.slabinfo']
```

This gives us some buffer protection against additional directory levels.

Reply:
That looks great, didn't know about this, results below.

# find /var/tmp/sosreport-jjansky-rhel8-2022-03-22-nhkqzps/ -iname "memory.kmem.slabinfo"
#

I will update PR with your suggestion.